### PR TITLE
Dev/block consolidation

### PR DESCRIPTION
**Pull request by nc6**
## Description

This builds upon #2863. When blocks are read they will additionally be converted to the consolidated blund format.

## Linked issue

https://iohk.myjetbrains.com/youtrack/issue/CDEC-293

### CODE REVIEW

**Comment by nc6** [comment]
This has been tested in the same fashion as #2863.

**Comment by erikd** [comment]
This looks pretty nice! I'm going to have a play with it.

File: block/src/Pos/DB/Block.hs [hunk 1]
```diff
@@ -272,21 +315,24 @@ deleteData fp = (liftIO $ removeFile fp) `catch` handler
...
-undoDataPath :: MonadRealDB ctx m => HeaderHash -> m FilePath
-undoDataPath hh = do
-    bdd <- view blockDataDir <$> getNodeDBs
-    pure $ (view _3) $ getAllPaths bdd hh
+-- | Paths at which we store the block data.
+data BlockStoragePaths = BlockStoragePaths
+  { bspRoot :: FilePath
+    -- | Block data itself.
+  , bspBlock :: FilePath
+    -- | Unfo information for a block.
```
Comment (SebastienGllmt) [inline block/src/Pos/DB/Block.hs]: Type: Should be "Undo" not "Unfo"

**Comment by erikd** [comment]
Working on getting proper QA testing done on this.

File: block/src/Pos/DB/Block.hs (hunk 1 above)
Comment (erikd) [reply block/src/Pos/DB/Block.hs]: Bingo! Will fix it.

**Comment by amias-iohk** [comment]
Tested with two builds 

legacy data files as made by daedalus-0.10.0-cardano-sl-1.1.1-mainnet-x86_64-linux-1362.bin
( https://buildkite.com/organizations/input-output-hk/pipelines/daedalus/builds/1362/jobs/cdf9a758-7d0b-4853-9854-983e5848ff48/artifacts/bb6733b1-ade5-4ccb-b206-3b4c8887f783 ) 

this pr was built in to  https://buildkite.com/input-output-hk/daedalus/builds/1409 

First - installed legacy build (i had no mainnet installs prior to test) and synced , then upgraded
verified old format data files (block + undo) recreated as blund files on upgrade 
this happens in the background and doesn't seem to break if daedalus is restarted 

Second - wiped mainnet folder and reran new installer 
verified fresh install creates blund files

**Comment by erikd** [comment]
Chatted to @amias-iohk in Slack. The above testing he did was on Linux. We may still decide to test on Mac and Windows before merging this.

**Comment by erikd** [comment]
:tada:

**Pull request merged**